src/Symfony/Component/Validator/Constraints/UniqueValidator.php

```diff
@@ -39,7 +39,7 @@ public function validate(mixed $value, Constraint $constraint): void
 
         $collectionElements = [];
         $normalizer = $this->getNormalizer($constraint);
-        foreach ($value as $element) {
+        foreach ($value as $index => $element) {
```

Comment:
Does this also work for instances of `IteratorAggregate`? We should at least add a test case for that as well.

Reply:
Added.